web_widget_date_interval/static/src/js/web_widget_date_interval.js

```diff
@@ -75,17 +75,22 @@ openerp.web_widget_date_interval = function (instance) {
                 }
                 return result;
             },
+            weeknumber_iso_date_from_week: function (year, week) {
+                return new Date(year, 5, 1).setWeek(week);
+            },
             weeknumber_iso_onchange_year: function (e) {
-                var year = parseInt(jQuery(e.currentTarget).data('year'), 10),
-                    value = this._get_date();
-                value.setFullYear(year);
-                return this.weeknumber_iso_update_interval(value);
+                var date = this._get_date();
+                var year = parseInt(jQuery(e.currentTarget).data('year'), 10);
+                var week = this.weeknumber_iso_weeknumber(date);
+                var monday = this.weeknumber_iso_date_from_week(year, week);
+                return this.weeknumber_iso_update_interval(monday);
             },
             weeknumber_iso_onchange_week: function (e) {
+                var date = this._get_date();
+                var year = date.clone().next().thursday().getFullYear();
```

Comment:
this gives you an off by one error if the date is on a Thursday or later. Better define `weeknumber_iso_yearnumber` which checks if the current date is a Thursday (do nothing) or later (use last Thursday).
Or change `get_date` to always ever return a Monday, then this should be solved too